addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    @api.model
+    def _map_all_eu_companies_taxes(self):
+        ''' Identifies EU companies and calls the _map_eu_taxes function
+        '''
+        eu_countries = self.env.ref('base.europe').country_ids
+        companies = self.search([('account_tax_fiscal_country_id','in', eu_countries.ids)])
+        companies._map_eu_taxes()
+
+    def _map_eu_taxes(self):
+        '''Creates or updates Fiscal Positions for each EU country excluding the company's account_tax_fiscal_country_id
+        '''
+        eu_countries = self.env.ref('base.europe').country_ids
+        oss_tax_groups = self.env['ir.model.data'].search([
+            ('module', '=', 'l10n_eu_service'),
+            ('model', '=', 'account.tax.group')])
+        for company in self:
+            invoice_repartition_lines, refund_repartition_lines = company._get_repartition_lines_oss()
+            taxes = self.env['account.tax'].search([
+                ('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id),
+                ('tax_group_id', 'not in', oss_tax_groups.mapped('res_id'))])
+            for country in eu_countries - company.account_tax_fiscal_country_id:
+                mapping = []
+                foreign_taxes = {}
+                fpos = self.env['account.fiscal.position'].search([
+                            ('country_id', '=', country.id),
+                            ('company_id', '=', company.id),
+                            ('auto_apply', '=', True),
+                            ('vat_required', '=', False)], limit=1)
+                if not fpos:
+                    fpos = self.env['account.fiscal.position'].create({
+                        'name': _('OSS B2C %s', country.name),
+                        'country_id': country.id,
+                        'company_id': company.id,
+                        'auto_apply': True,
+                    })
+                for domestic_tax in taxes:
+                    tax_amount = EU_TAX_MAP.get((company.account_tax_fiscal_country_id.code, domestic_tax.amount, country.code), False)
+                    if tax_amount and not domestic_tax in fpos.tax_ids.tax_src_id:
+                        if not foreign_taxes.get(tax_amount, False):
+                            if not self.env['ir.model.data'].xmlid_to_object('l10n_eu_service.oss_tax_group_%s' % tax_amount):
```

Comment:
Maybe format the tax amount to replace the decimal "." by "_". Usually, xmlids don't contain ".", except after the module name. It shouldn't break anything to keep it as-is, but I'm afraid some deep part of the code somewhere could consider xmlids only have . after module name.